src/Catalyst.Core.Modules.Dfs/UnixFs/ChunkedStream.cs

```diff
@@ -0,0 +1,179 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Core.Lib.Dag;
+using Lib.P2P;
+using ProtoBuf;
+
+namespace Catalyst.Core.Modules.Dfs.UnixFs
+{
+    /// <summary>
+    ///   Provides read-only access to a chunked file.
+    /// </summary>
+    /// <remarks>
+    ///   Internal class to support <see cref="UnixFs"/>.
+    /// </remarks>
+    public sealed class ChunkedStream : Stream
+    {
+        private sealed class BlockInfo
+        {
+            public Cid Id;
+            public long Position;
+        }
+
+        private List<BlockInfo> blocks = new List<BlockInfo>();
+        private long fileSize;
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="ChunkedStream"/> class with
+        ///   the specified <see cref="IBlockApi"/> and <see cref="DagNode"/>.
+        /// </summary>
+        /// <param name="blockService"></param>
+        /// <param name="keyChain"></param>
+        /// <param name="dag"></param>
+        internal ChunkedStream(IBlockApi blockService, IKeyApi keyChain, IDagNode dag)
+        {
+            BlockService = blockService;
+            KeyChain = keyChain;
+            var links = dag.Links.ToArray();
+            var dm = Serializer.Deserialize<DataMessage>(dag.DataStream);
+            if (dm.FileSize != null) fileSize = (long) dm.FileSize;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)